### PR TITLE
initial checkin to support rds version upgrade

### PR DESCRIPTION
**Pull request by bjeevan-ib**
*support for db version upgrade
* fixed password escape issue where some of the symbols were not getting encoded/decoded as expected
* removed pg_cron from extension temporarily as this was impacting upgrade

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/infobloxopen/db-controller
 
-go 1.17
+go 1.19
```
Comment (kd7lxl) [inline go.mod]: Don't forget to update https://github.com/infobloxopen/db-controller/blob/b9cdbe510ed7f65146ca08eb5836d67b24ad36a1/Jenkinsfile#L8

Comment (kd7lxl) [reply go.mod]: And https://github.com/infobloxopen/db-controller/blob/b9cdbe510ed7f65146ca08eb5836d67b24ad36a1/Dockerfile#L2

Comment (bjeevan-ib) [reply go.mod]: Done.

File: controllers/databaseclaim_controller.go [hunk 2]
```diff
@@ -584,6 +589,7 @@ loop:
 
 	//done with migration- switch active server to newDB
 	dbClaim.Status.ActiveDB = *dbClaim.Status.NewDB.DeepCopy()
+	dbClaim.Status.ActiveDB.Status = "ready"
```
Comment (kd7lxl) [inline controllers/databaseclaim_controller.go:592]: Seems like there should be a type and const for this. Like this: https://github.com/Infoblox-CTO/s3-controller/blob/fb9a5fa1fc690ed7e6ab029c38992cde7ef91cba/api/v1/bucketclaim_types.go#L65-L83

File: controllers/databaseclaim_controller.go [hunk 3]
```diff
@@ -733,14 +740,22 @@ func (r *DatabaseClaimReconciler) generatePassword() (string, error) {
 	minPasswordLength := r.getMinPasswordLength()
 	complEnabled := r.isPasswordComplexity()
 
+	// Customize the list of symbols.
```
Comment (kd7lxl) [inline controllers/databaseclaim_controller.go:743]: I can see that this code customizes the list of symbols, but what would be helpful for the comment to answer is _why_ it's necessary to customize the list.

File: pkg/dbclient/client.go [hunk 4]
```diff
@@ -142,26 +138,29 @@ func (pc *client) CreateDatabase(dbName string) (bool, error) {
...
-			pc.log.Info("created extension " + s)
-		}
 	}
 
 	return created, err
 }
 
+func (pc *client) CreateDefaultExtentions(dbName string) error {
+	db, err := pc.getDB(dbName)
+	pc.log.Info("connected to " + dbName)
```
Comment (kd7lxl) [inline pkg/dbclient/client.go:148]: Doesn't really make sense to log "connected to" before checking if there was a connection error.

File: pkg/dbclient/client_test.go [hunk 5]
```diff
@@ -98,8 +98,9 @@ func setupSqlDB(t *testing.T) *testDB {
 
 	// Exponential retry to connect to database while it is booting
 	if err := pool.Retry(func() error {
-		dbConnStr := fmt.Sprintf("host=%s port=%d user=%s dbname=postgres password=%s sslmode=disable",
-			host, port, user, pass)
+		//dbConnStr := fmt.Sprintf("host=%s port=%d user=%s dbname=postgres password=%s sslmode=disable",
+		//	host, port, user, pass)
```
Comment (kd7lxl) [inline pkg/dbclient/client_test.go:102]: Why is this commented out? Can it be removed?

Comment (bjeevan-ib) [reply pkg/dbclient/client_test.go:102]: done

File: controllers/databaseclaim_controller.go [hunk 6]
```diff
@@ -733,14 +740,23 @@ func (r *DatabaseClaimReconciler) generatePassword() (string, error) {
...
+	// Removed \ from the default list as the encoding/decoding was treating it as an escape character
```
Comment (kd7lxl) [inline controllers/databaseclaim_controller.go:744]: Super helpful, thank you 👍

File: pkg/dbclient/client_test.go (hunk 5 above)
Comment (kd7lxl) [reply pkg/dbclient/client_test.go:102]: Are you going to clean it up?

Comment (bjeevan-ib) [reply pkg/dbclient/client_test.go]: @kd7lxl yikes - was looking at the wrong file.(client.go instead of client_test.go). removed it now.

**Review by kd7lxl — APPROVED** [review]

**Pull request merged**